bioblend/galaxy/objects/wrappers.py

```diff
@@ -183,6 +183,7 @@ class Workflow(Wrapper):
     def __init__(self, wf_dict, gi=None):
         super(Workflow, self).__init__(wf_dict, gi=gi)
         missing_ids = []
+        tools_list_by_id = [t.id for t in gi.tools.list()]
```

Comment:
Can you please `s/gi.tools.list()/gi.tools.get_previews()/` ?

Presently they are the same function, but in the future `list()` may be more expensive to evaluate.